scipy/sparse/_data.py

```diff
@@ -107,12 +107,23 @@ def power(self, n, dtype=None):
 
         Parameters
         ----------
-        n : n is a scalar
+        n : scalar
+            n is a non-zero scalar (nonzero avoids dense ones creation)
+            If zero power is desired, special case it to use `np.ones`
 
         dtype : If dtype is not specified, the current dtype will be preserved.
+
+        Raises
+        ------
+        ValueError : if n is a zero scalar
+            If zero power is desired, special case it to use
+            `np.ones(A.shape, dtype=A.dtype)`
```

Comment:
Think this is equivalent 
```suggestion
            `np.ones_like(A)`
```

Reply:
I don’t think that works… the sparse arrays don’t have `__array_function__` or `__array__` so they show up as a scalar Python object and `ones_like(A)` gives ` np.array(1)`.  Would love to hear why scipy.sparse doesn’t use __array__ and friends, but I’m sure there are good reasons.  Anyway, we can’t use `ones_like` on a sparse array. But we can use `ones with the shape passed in.

Reply:
Indeed, my oversight